packages/js-sdk/src/Dm3Sdk.test.ts

```diff
@@ -1,34 +1,278 @@
-import { StorageAPI } from '@dm3-org/dm3-lib-storage';
+
+import MockAdapter from 'axios-mock-adapter';
+import { normalizeEnsName } from '@dm3-org/dm3-lib-profile';
+import { ITLDResolver } from './tld/nameService/ITLDResolver';
 
 describe('Dm3Sdk', () => {
-    let upController: ethers.Signer;
+    let alice: MockedUserProfile;
+    let bob: MockedUserProfile;
+    let karl: MockedUserProfile;
+
+    //Axios mock to mock the http requests
+    let axiosMock;
+
+    let deliveryService: MockDeliveryServiceProfile;
 
     beforeEach(async () => {
-        upController = ethers.Wallet.createRandom();
+        alice = await mockUserProfile(
+            ethers.Wallet.createRandom(),
+            'alice.up',
+            ['test.io'],
+        );
+        bob = await mockUserProfile(ethers.Wallet.createRandom(), 'bob.up', [
+            'test.io',
+        ]);
+
+        karl = await mockUserProfile(ethers.Wallet.createRandom(), 'karl.up', [
+            'test.io',
+        ]);
+
+        deliveryService = await getMockDeliveryServiceProfile(
+            ethers.Wallet.createRandom(),
+            'http://localhost:3000',
+        );
+
+        axiosMock = new MockAdapter(axios);
+
+        //Mock BackendConnector HttpRequests
+        //Mock profileExistsOnDeliveryService
+        axiosMock
+            .onGet(
+                `http://localhost:4060/profile/${normalizeEnsName(
+                    alice.address,
+                )}.addr.test`,
+            )
+            .reply(200);
+        //Mock getChallenge
+        axiosMock
+            .onGet(
+                `http://localhost:4060/auth/${normalizeEnsName(
+                    alice.address,
+                )}.addr.test`,
+            )
+            .reply(200, 'mock-challenge');
+
+        //Mock getToken
+        axiosMock
+            .onPost(
+                `http://localhost:4060/auth/${normalizeEnsName(
+                    alice.address,
+                )}.addr.test`,
+            )
+            .reply(200);
+    });
+
+    describe('conversations', () => {
+        it('can add a conversation to the contact list', async () => {
+            const mockTldResolver = {
+                resolveTLDtoAlias: async () =>
+                    `${normalizeEnsName(bob.address)}.addr.test`,
+                resolveAliasToTLD: async () => 'bob.eth',
+            } as unknown as ITLDResolver;
+
+            const mockConfig: Dm3SdkConfig = {
+                mainnetProvider: {} as ethers.providers.JsonRpcProvider,
+                storageApi: {
+                    addConversation: async () => {},
+                } as unknown as StorageAPI,
+                nonce: '1',
+                defaultDeliveryService: 'test.io',
+                addressEnsSubdomain: '.addr.test',
+                userEnsSubdomain: '.user.test',
+                resolverBackendUrl: 'resolver.io',
+                backendUrl: 'http://localhost:4060',
+                _tld: mockTldResolver,
```

Comment:
if the only purpose of `_tld` in config is to mock the function, we could also probably use jest mocking instead.
just to keep in mind for future refactorings.

https://jestjs.io/docs/mock-function-api#mockfnmockimplementationfn:~:text=jest.mocked(SomeClass).mockImplementation(()%20%3D%3E%20%7B%0A%20%20return%20%7B%0A%20%20%20%20method%3A%20mockMethod%2C%0A%20%20%7D%3B%0A%7D)%3B

Reply:
Thats an intresting though. I don't see a reason why tld should be overwritten. However given the object might enhance flexibility. 
In that case jest mock could be an interesting option to reduce complexity